bokeh/models/plots.py

```diff
@@ -239,6 +240,31 @@ def add_glyph(self, source_or_glyph, glyph=None, **kw):
         self.renderers.append(g)
         return g
 
+    def add_annotation(self, annotation, **kwargs):
+        '''Adds new Annotation into the Plot.renderers
+
+        Args:
+            annotation (Annotation) : instance of annotation to add to plot
+        Returns:
+            Annotation
+
+        '''
+        if not isinstance(annotation, Annotation):
+            raise ValueError("'annotation' argument to add_annotation must be \
+                             Annotation subclass")
+
+        if annotation.plot is not None:
+            raise ValueError("annotation %s to be added already has 'plot' \
+                             attribute set" % annotation)
+
+        if hasattr(annotation, 'source') and annotation.source is None:
+            annotation.source = ColumnDataSource()
```

Comment:
Does this need to be handled here? We can add a validation to check for missing source/columns if field names are given, and then just send the `None` on to BokehJS deal with. Alternatively we can make the default for `source` on this model be a lambda that returns a new CDS